pkg/operation/botanist/component/vpnshoot/vpnshoot_test.go

```diff
@@ -0,0 +1,848 @@
+				Expect(string(managedResourceSecret.Data["verticalpodautoscaler__kube-system__vpn-shoot.yaml"])).To(Equal(vpaYAML))
+				Expect(string(managedResourceSecret.Data["secret__kube-system__"+secretNameTest+".yaml"])).To(Equal(secretYAML))
+				Expect(string(managedResourceSecret.Data["deployment__kube-system__vpn-shoot.yaml"])).To(Equal(deploymentYAML))
+			})
+		})
+
+		Describe("#Destroy", func() {
+			It("should successfully destroy all resources", func() {
+				Expect(c.Create(ctx, managedResource)).To(Succeed())
+				Expect(c.Create(ctx, managedResourceSecret)).To(Succeed())
+
+				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(Succeed())
+
+				Expect(vpnShoot.Destroy(ctx)).To(Succeed())
+
+				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(MatchError(apierrors.NewNotFound(schema.GroupResource{Group: resourcesv1alpha1.SchemeGroupVersion.Group, Resource: "managedresources"}, managedResource.Name)))
+				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceSecret), managedResourceSecret)).To(MatchError(apierrors.NewNotFound(schema.GroupResource{Group: corev1.SchemeGroupVersion.Group, Resource: "secrets"}, managedResourceSecret.Name)))
+			})
+		})
+
+		deploymentYAML = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    ` + references.AnnotationKey(references.KindSecret, secretNameTest) + `: ` + secretNameTest + `
+    ` + references.AnnotationKey(references.KindSecret, secretNameTLSAuthTest) + `: ` + secretNameTLSAuthTest + `
+  creationTimestamp: null
+  labels:
+    app: vpn-shoot
+    gardener.cloud/role: system-component
+    origin: gardener
+  name: vpn-shoot
+  namespace: kube-system
+spec:
+  replicas: 1
+  revisionHistoryLimit: 1
+  selector:
+    matchLabels:
+      app: vpn-shoot
+  strategy:
+    rollingUpdate:
+      maxSurge: 100%
+      maxUnavailable: 0%
+    type: RollingUpdate
+  template:
+    metadata:
+      annotations:
+        ` + references.AnnotationKey(references.KindSecret, secretNameTest) + `: ` + secretNameTest + `
+        ` + references.AnnotationKey(references.KindSecret, secretNameTLSAuthTest) + `: ` + secretNameTLSAuthTest + `
+      creationTimestamp: null
+      labels:
+        app: vpn-shoot
+        gardener.cloud/role: system-component
+        origin: gardener
+        type: tunnel
+    spec:
+      automountServiceAccountToken: false
+      containers:
+      - env:
+        - name: SERVICE_NETWORK
+          value: ` + serviceNetwork + `
+        - name: POD_NETWORK
+          value: ` + podNetwork + `
+        - name: NODE_NETWORK
+          value: ` + nodeNetwork + `
+        - name: ENDPOINT
+          value: ` + endPoint + `
+        - name: OPENVPN_PORT
+          value: "8132"
```

Comment:
This  field is a string. When I initialise it as a string, value = "8132" and pass it to the deployment, in the managedresource secret it appears as "8132" with the quotes. So In the test also I included the quotes even though in the original YAML it was just 8132. 
Is it fine?

Reply:
We should include the quotes here, yes. Otherwise this is no valid manifest. Actually, in the original YAML it must have been with quotes as well, otherwise it would not have worked.